src/optimization/steepest_descent_method.cpp

```diff
@@ -0,0 +1,302 @@
+#include "k52/optimization/steepest_descent_method.h"
+
+#ifdef BUILD_WITH_MPI
+
+#include <boost/mpi.hpp>
+#include <k52/parallel/mpi/constants.h>
+
+#endif
+
+#include <iostream>
+#include <cmath>
+
+#include <k52/optimization/params/i_continuous_parameters.h>
+
+using ::std::vector;
+
+namespace k52
+{
+namespace optimization
+{
+
+SteepestDescentMethod::SteepestDescentMethod(
+	double alpha,
+	double increment_of_the_argument,
+	size_t max_iteration_number,
+	double precision)
+	: alpha_(alpha)
+	, increment_of_the_argument_(increment_of_the_argument)
+	, max_iteration_number_(max_iteration_number)
+	, precision_(precision)
+{
+}
+
+SteepestDescentMethod* SteepestDescentMethod::Clone() const
+{
+	return new SteepestDescentMethod(
+		alpha_,
+		increment_of_the_argument_,
+		max_iteration_number_,
+		precision_);
+}
+
+std::string SteepestDescentMethod::get_name() const
+{
+	return "Steepest Descent Method";
+}
+
+
+#ifdef BUILD_WITH_MPI
+void SteepestDescentMethod::Send(boost::mpi::communicator* communicator, int target) const
+ {
```

Comment:
alfa is not transmitted, but it will be ok if it will be const (if it is a proven optimal value - need a link to proof source)